src/aktualizr_repo/main.cc

```diff
@@ -68,7 +71,14 @@ int main(int argc, char **argv) {
         repo.generateRepo(key_type);
       } else if (command == "image") {
         if (vm.count("filename") > 0) {
-          repo.addImage(vm["filename"].as<std::string>());
+          Delegation delegation;
+          if (vm.count("dname") != 0) {
+            delegation = Delegation(vm["path"].as<boost::filesystem::path>(), vm["dname"].as<std::string>());
+            if (!delegation.isMatched(vm["filename"].as<boost::filesystem::path>())) {
+              throw std::runtime_error("Image path doesn't match delegation!");
+            }
+          }
+          repo.addImage(vm["filename"].as<boost::filesystem::path>(), delegation);
```

Comment:
For now, we need to set `terminating` to `true`. That should be the default, but it would also be acceptable to explicitly set it to `true` here.